[DOCS-10227] VS Code extension revamp (#28826)
DataDog/documentation: content/en/developers/ide_plugins/vscode/_index.md

---

> title: Datadog Extension for VS Code, Cursor, and Related Forks

Reviewer: I guess we can remove the word "Datadog" if it's too overloaded as is? Also maybe "Related" is not useful here?

> Extension for VS Code, Cursor and Other Forks

Reviewer 2: To add another view-point, I would go with `Datadog Extension for VS Code & Cursor`.  Those are the only two flavors we are officially backing at the moment, and by providing the Open VSX Registry info, we already open the door to others who want to experiment unofficially with other flavors (Windsurf etc).

- [**Log Annotations**](#log-annotations): Gauge log volumes and search logs from your code.

- [**Code Insights**](#code-insights): Stay informed about runtime errors, vulnerabilities, and flaky tests without leaving the code.

- [**View in IDE**](#view-in-ide): Jump directly from code references in Datadog to your source files.

- [**Static Code Analysis**](#static-code-analysis): Detect and fix problems before you commit changes.

- [**Exception Replay**](#exception-replay): Debug your production code.

<div class="alert alert-info">Unless stated otherwise, all extension features are available for both VS Code and any other IDEs based on VS Code forks, such as Cursor.</div>

## Requirements

- **Datadog account**: Most features require a Datadog account.  
  - New to Datadog? [Learn more][3] about Datadog’s monitoring and security tools and sign up for a free trial.  
  - If your organization uses a custom sub-domain such as `myorg.datadoghq.com`, you must indicate it using the `datadog.connection.oauth.setup.domain` setting.

- **Git**: The extension works better when Git is enabled in the IDE. Ensure this is enabled by checking the `git.enabled` setting.

## Log annotations

> Use **Log Annotations** to gauge the volume of logs generated by a given log line in your code. The extension adds CodeLens annotations to detected logging patterns that match log records in Datadog. Click an annotation to open the [Log Explorer][4] in Datadog and view the matching logs.

Reviewer: "CodeLens" is not something devs really know about, it's a VS Code branded term. 
I'd replace:
> The extension adds CodeLens annotations to detected logging patterns that match log records in Datadog.

with 
> The extension adds annotations above your code to detected logging patterns that match log records in Datadog.

Author: ```suggestion
Use **Log Annotations** to gauge the volume of logs generated by a given log line in your code. The extension adds annotations above your code to detected logging patterns that match log records in Datadog. Click an annotation to open the [Log Explorer][4] in Datadog and view the matching logs.
```

The extension identifies errors and vulnerabilities in the code with colored squiggles; hover over the line for more details.

{{< img src="/developers/ide_plugins/vscode/code-insights-inline-hover.mp4" alt="Hovering over inline code insights" style="width:100%" video=true >}}

The **Code Insights** view in the Datadog sidebar lists all the issues found in the repository. Select an item to view the full insight, and use the links to jump to the related source code location or open the code insight in Datadog.

You can group code insights by kind, file, priority, or service. You can also ignore individual code insights and set filters to view the ones you're most interested in.

{{< img src="/developers/ide_plugins/vscode/code-insights-2.png" alt="The Code Insights view." style="width:100%;" >}}

For specific insights about the file currently open in the active editor, check the **File Insights** view in the IDE's file explorer. This view also lists issues discovered by [Static Code Analysis](#static-code-analysis) within the file.

{{< img src="/developers/ide_plugins/vscode/file_insights_view.mp4" alt="File Insights view" style="width:100%" video=true >}}

> ## View in IDE

Reviewer: This feature only works for VS Code and Cursor, it doesn't work for other forks (since we need to manually add the entry in the menu for every new IDE, which means new icons and more importantly takes up valuable real estate in an already crammed menu). I'd mention it as a callout (alert-info).

To access this feature, you must enable [Error Tracking Exception Replay][11] on Datadog.

After the feature has been enabled, you can see an **Exception Replay** button next to the stack trace section of any instrumented Error Tracking code insight. Click the button to:

- Access all the information Datadog has about the different frames.
- Navigate through the production code.
- Review the value of the different variables involved.

Select an Error Tracking code insight from the Code Insights view. Go to the stack trace and click the Exception Replay button. The IDE shows a new activity with two new views:

- **Variables**: Displays the variables related to a particular stack trace frame.
- **Stack Trace**: Lists the stack frames for navigation.

Select a stack trace frame and inspect the values of all the variables that Datadog captured from your production code.

{{< img src="/developers/ide_plugins/vscode/exception_replay.mp4" alt="Preview of Exception Replay" style="width:100%" video=true >}}

## License

Read the [End-User License Agreement][12] carefully before downloading or using the Datadog Visual Studio Code extension.

## Data and telemetry

> Datadog anonymously collects information about your usage of this IDE, including how you interact with it, whether errors occurred while using it, and what caused those errors, in accordance with the [Datadog Privacy Policy][13] and Datadog's [VS Code extension EULA][12].

Reviewer: Remove "anonymously", as we're about to de-anonymize the telemetry

> Datadog anonymously collects information...

should be: 
>Datadog collects information...

Author: @Vomain looks like this still has a placeholder `[List the identifiers that are being collected]` — what should we put in its place?

Reviewer: @joepeeples `user identifiers`

Author: Revised in 73e6dc5616e59bcaf8dda602ed7501251ff27fb5

> If you don't wish to send this data to [Datadog][3], you can opt out at any time in the extension's settings: `Datadog > Telemetry > Setup > Enable Telemetry` and select `disabled`.

Reviewer: > If you don't wish to send this data to [Datadog][3], you can disable this at any time...

Author: ```suggestion
If you don't wish to send this data to [Datadog][3], you can disable this at any time in the extension's settings: `Datadog > Telemetry > Setup > Enable Telemetry` and select `disabled`.
```

- [Visual Studio Marketplace][1]: Install the extension for VS Code.
- [Open VSX Registry][2]: Download the VSIX file to install for Cursor and other forks of VS Code.

{{< img src="/developers/ide_plugins/vscode/datadog-vscode-3.png" alt="The Datadog for VS Code extension" style="width:100%;" >}}

The extension includes these features:

- [**Log Annotations**](#log-annotations): Gauge log volumes and search logs from your code.

- [**Code Insights**](#code-insights): Stay informed about runtime errors, vulnerabilities, and flaky tests without leaving the code.

- [**View in IDE**](#view-in-ide): Jump directly from code references in Datadog to your source files.

- [**Static Code Analysis**](#static-code-analysis): Detect and fix problems before you commit changes.

- [**Exception Replay**](#exception-replay): Debug your production code.

<div class="alert alert-info">Unless stated otherwise, all extension features are available for both VS Code and any other IDEs based on VS Code forks, such as Cursor.</div>

## Requirements

- **Datadog account**: Most features require a Datadog account.  

>   - New to Datadog? [Learn more][3] about Datadog’s monitoring and security tools and sign up for a free trial.  

Author: ```suggestion
  - New to Datadog? [Learn more][3] about Datadog's monitoring and security tools and sign up for a free trial.  
```

## Static Code Analysis

[Static Code Analysis][8] analyzes your code (locally) against predefined rules to detect and fix problems.

The extension runs Static Code Analysis rules on the source files you have open in your workspace. This allows you to detect and fix problems such as maintainability issues, bugs, or security vulnerabilities in the code before you commit your changes.

Static Code Analysis supports scanning for many programming languages. For a complete list, see [Static Code Analysis Rules][9]. For file types belonging to supported languages, issues are shown in the source code editor, and you can directly apply suggested fixes.

{{< img src="/developers/ide_plugins/vscode/static_analysis.mp4" alt="Preview of Static Analysis" style="width:100%" video=true >}}

### Get started with Static Code Analysis

When you start editing a source file, the extension checks for [`static-analysis.datadog.yml`][10] at your source repository's root. It prompts you to create it if necessary.

{{< img src="/developers/ide_plugins/vscode/static-analysis-onboard.png" alt="Onboarding banner for setting up Static Code Analysis with Python files" style="width:75%;" >}}

> After you create the configuration file, the analyzer runs automatically in the background whenever you open a file. If you need to enable Static Code Analysis for a particular language, use the `Datadog: Configure Static Analysis Languages` command from the command palette (`Shift` + `Cmd` + `P`, then search for the command).

Reviewer 3: Might help to have the Windows equivalent shortcut too. (Ctrl + Shift + P.)

Author: Added in 8f8ead49ec77974d01cb0a66a2865d98cbe0397d
